src/components/app.jsx

```diff
@@ -1,9 +1,74 @@
-import React from "react";
-import { ReactComponent as Logo } from "../images/sun.svg";
+import React, { useState, useEffect } from "react";
 import "../App.css";
+import CurrentWeather from "./CurrentWeather";
+import FiveDayForecast from "./FiveDayForecast";
+import Nav from "./Nav";
+import UnitCheckbox from "./UnitCheckbox";
 
 function App() {
-  return <h1>Weather</h1>;
+  const [currentView, setCurrentView] = useState("currentWeather");
+  const [lat, setLat] = useState(null);
+  const [lon, setLon] = useState(null);
+  const [err, setErr] = useState(null);
+  const [unit, setUnit] = useState("imperial");
+
+  const OPEN_WEATHER_APP_ID = "62536724de969bf01c1e02a249c958f9";
+
+  useEffect(() => {
+    // Make a one time call to our browser to get current position
+    window.navigator.geolocation.getCurrentPosition(
+      (position) => {
+        setLat(position.coords.latitude);
+        setLon(position.coords.longitude);
+      },
+      (err) => {
+        setErr(err);
+      }
+    );
+  }, []);
+
+  const renderContent = () => {
+    if (err) {
+      return "Whoops, an error occurred. Have you allowed your browser to access your location?";
+    }
+
+    if (!lon || !lat) {
+      return (
+        <div class="ui active dimmer">
+          <div class="ui text loader">Please wait...</div>
+        </div>
+      );
+    }
+
+    if (currentView === "currentWeather") {
```

Comment:
I can see the benefit of keeping data in CurrentWeather component refreshed. For example, if a user was looking at this CurrentWeather view at 8am and then switched to 5 Day view and went off to do something else for a while, if that user then switches back to CurrentWeather let's say at 8:30am or 9am, then the current weather temperature could have changed. I can't say if that is a common use case.

A possible problem could be that since 5 day forecast, since it only shows daily Highs and Lows, should not change given the same days, it's not that efficient to make a new network request every time we mount the component. A possible workaround is to add some sort of caching or lazy loading of the 5 day forecast data. Here, I'm also assuming location never changes.